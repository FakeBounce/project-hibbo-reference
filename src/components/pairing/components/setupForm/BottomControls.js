import React from 'react';
import { Animated } from 'react-native';
import {
  string,
  func,
  oneOfType,
  object,
  array,
  number,
  shape,
  instanceOf,
} from 'prop-types';
import styled from 'styled-components/native';

import theme from 'styles/appStyles';

import { StyledTextSmall } from 'styledComponents/texts';

import * as Styles from './styles';

const SText = styled(Animated.createAnimatedComponent(StyledTextSmall))`
  background-color: transparent;
`;

const BottomControls = ({
  leftLabel,
  leftAction,
  rightLabel,
  rightAction,
  style,
  animations: { left, right },
}) => {
  const inputRange = [0, 1];
  const outputRange = [theme.colors.white, theme.colors.yellow];
  const leftButtonColor = left.colorInterpolate.interpolate({
    inputRange,
    outputRange,
  });
  const rightButtonColor = right.colorInterpolate.interpolate({
    inputRange,
    outputRange,
  });

  return (
    <Styles.AContainerControls style={style}>
      {leftLabel !== '' && (
        <Styles.SButtonControls
          activeOpacity={1}
          onPress={() => {
            leftAction();
          }}
          style={{
            opacity: left.opacity,
            transform: [{ translateY: left.translateY }, { scale: left.scale }],
          }}
        >
          <SText style={{ color: leftButtonColor }}>{leftLabel}</SText>
        </Styles.SButtonControls>
      )}

      <Styles.SButtonControls
        activeOpacity={1}
        onPress={() => {
          rightAction();
        }}
        style={{
          opacity: right.opacity,
          transform: [{ translateY: right.translateY }, { scale: right.scale }],
        }}
      >
        <SText style={{ color: rightButtonColor }}>{rightLabel}</SText>
      </Styles.SButtonControls>
    </Styles.AContainerControls>
  );
};

BottomControls.defaultProps = {
  leftAction: () => {},
  leftLabel: '',
  style: {},
};

BottomControls.propTypes = {
  leftLabel: string,
  leftAction: func,
  rightLabel: string.isRequired,
  rightAction: func.isRequired,
  style: oneOfType([object, number, array]),
  animations: shape({
    left: shape({
      opacity: instanceOf(Animated.Value).isRequired,
      translateY: instanceOf(Animated.Value).isRequired,
      scale: instanceOf(Animated.Value).isRequired,
    }).isRequired,
    right: shape({
      opacity: instanceOf(Animated.Value).isRequired,
      translateY: instanceOf(Animated.Value).isRequired,
      scale: instanceOf(Animated.Value).isRequired,
    }).isRequired,
  }).isRequired,
};

export default BottomControls;
