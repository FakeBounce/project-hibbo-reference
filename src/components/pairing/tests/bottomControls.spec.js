import React from 'react';
import renderer from 'react-test-renderer';

import BottomControls from '../components/setupForm/BottomControls';

describe('BottomControls', () => {
  const fn = jest.fn();
  const component = renderer.create(
    <BottomControls
      leftLabel=""
      leftAction={fn}
      RightLabel=""
      RightAction={fn}
      style={{}}
    />,
  ).toJSON();

  it('should render the component', () => {
    expect(component).toMatchSnapshot();
  });
});
