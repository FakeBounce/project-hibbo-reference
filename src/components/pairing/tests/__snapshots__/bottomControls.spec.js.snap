// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BottomControls should render the component 1`] = `
<View
  collapsable={undefined}
  style={Object {}}
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={undefined}
    >
      
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={undefined}
    />
  </View>
</View>
`;
